geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLTypes.scala

```diff
@@ -41,7 +41,9 @@ object SQLTypes {
   UDTRegistration.register(classOf[Geometry].getCanonicalName, classOf[GeometryUDT].getCanonicalName)
 
   def init(sqlContext: SQLContext): Unit = {
+    SQLGeometricCastFunctions.registerFunctions(sqlContext)
```

Comment:
I personally put my registerFunction call within the `SQLSpatialFunctions.registerFunctions(sqlContext)` function. Don't which approach makes the most sense.